src/Primitive.ts

```diff
@@ -34,7 +34,9 @@ export type Primitive<T> = Equal<T, PrimitiveMain<T>> extends true
 
 type Equal<X, Y> = X extends Y ? (Y extends X ? true : false) : false;
 
-type PrimitiveMain<Instance> = ValueOf<Instance> extends object
+type PrimitiveMain<Instance> = Instance extends [never]
```

Comment:
Personally, I think it will be difficult for developers in the distant future to know that this is a part made to respond to union types such as Date | null. Wouldn't it be better to add comments?